nvflare/private/fed/client/fed_client_base.py

```diff
@@ -138,7 +137,8 @@ def set_sp(self, project_name, sp: SP):
 
     def _switch_ssid(self):
         if self.engine:
-            self.engine.abort_task(self.engine.run_number)
+            for run_number in self.engine.client_executor.run_processes.keys():
```

Comment:
This line is assuming, if self.engine exists then it has a variable call "client_executor" and that exists and that client_executor has a variable call "run_processes".

I think we should instead define an API in the engine spec.
And we need to check if this self.engine is of that type and then call the corresponding method.